Rubberduck.Parsing/ParserRuleContextExtensions.cs

```diff
@@ -164,6 +164,111 @@ private static TContext GetAncestor_Recursive<TContext>(ParserRuleContext contex
             return listener.Matches;
         }
 
+        public static IEnumerable<T> GetChildren<T>(this RuleContext context)
+        {
+            if (context == null)
+            {
+                yield break;
+            }
+
+            for (var index = 0; index < context.ChildCount; index++)
+            {
+                var child = context.GetChild(index);
+                if (child is T)
+                {
+                    yield return (T)child;
+                }
+            }
+        }
+
+        public static bool HasParent(this RuleContext context, RuleContext parent)
+        {
+            if (context == null)
+            {
+                return false;
+            }
+            if (context == parent)
+            {
+                return true;
+            }
+            return HasParent(context.Parent, parent);
+        }
+
+        public static TContext FindChild<TContext>(this ParserRuleContext context) where TContext : ParserRuleContext
+        {
+            if (context == null)
+            {
+                return default;
+            }
+
+            for (var index = 0; index < context.ChildCount; index++)
+            {
+                var child = context.GetChild(index);
+                if (context.GetChild(index) is TContext)
+                {
+                    return (TContext)child;
+                }
+            }
+            return default;
+        }
+
+        public static bool HasChildToken(this IParseTree context, string token)
+        {
+            for (var index = 0; index < context.ChildCount; index++)
+            {
+                var child = context.GetChild(index);
+                if (context.GetChild(index).GetText().Equals(token))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public static T GetDescendent<T>(this IParseTree context)
```

Comment:
I'm pretty sure I saw this one before... was it in a retracted PR by you?

Reply:
Yes, I think so.

Reply:
Method removed